### PR TITLE
Feature/fix utils

### PR DESCRIPTION
**Pull request by shuds13**
parse_args ignores unrecognized arguments, allowing user to have positional args in calling script.
utils.py uses a logger instead of print statements (xsdk).
Added/updated submission script examples.
Various docs additions and cleanup.

### CODE REVIEW

File: docs/platforms/example_scripts.rst [hunk 1]
```diff
@@ -26,9 +26,23 @@ Blues
     :caption: /examples/job_submission_scripts/blues_script.pbs
     :language: bash
 
-Theta - Central Mode with Balsam
+Theta - On MOM nodes with multiprocessing
```
Comment (jmlarson1) [inline docs/platforms/example_scripts.rst]: Should it be "nodes" or "node"?

File: libensemble/tests/scaling_tests/forces/theta_submit_balsam.sh [hunk 2]
```diff
@@ -92,7 +92,7 @@ balsam launcher --consume-all --job-mode=mpi --num-transition-threads=1
 if [[ $LIBE_PLOTS = "true" ]]; then
   python $PLOT_DIR/plot_libe_calcs_util_v_time.py
   python $PLOT_DIR/plot_libe_runs_util_v_time.py
-  python $PLOT_DIR/plot_libE_histogram.pyfi
+  python $PLOT_DIR/plot_libe_histogram.pyfi
```
Comment (jmlarson1) [inline libensemble/tests/scaling_tests/forces/theta_submit_balsam.sh]: `.pyfi`?

File: docs/platforms/example_scripts.rst (hunk 1 above)
Comment (shuds13) [reply docs/platforms/example_scripts.rst]: Yes, I'm fixing

File: libensemble/tests/scaling_tests/forces/theta_submit_balsam.sh (hunk 2 above)
Comment (shuds13) [reply libensemble/tests/scaling_tests/forces/theta_submit_balsam.sh]: Well spotted, I'm fixing.

**Pull request merged**